send/bbmri_collections

```diff
@@ -31,25 +31,40 @@ USER_TRANSPORT_COMMAND="$TRANSPORT_COMMAND -d @- $USERNAME_AND_PASSWORD $DESTINA
 GROUP_TRANSPORT_COMMAND="$TRANSPORT_COMMAND -d @- $USERNAME_AND_PASSWORD $DESTINATION$GROUP_ENDPOINT"
 
 # HTTP Request with timeout for USERS
-cat "$USER_FILE" | timeout -k $TIMEOUT_KILL $TIMEOUT $USER_TRANSPORT_COMMAND
+jq -c '.' "$USER_FILE" | timeout -k $TIMEOUT_KILL $TIMEOUT $USER_TRANSPORT_COMMAND
 # Catch errors
-ERR_CODE=$?
-#Error code 124 means - timed out
-if [ $ERR_CODE -eq 124 ]; then
-	echo "Communication with the peer has timed out with return code: $ERR_CODE (Warning: this error can mask original error 124 from peer!)" >&2
+# Catch errors
+RES_CODE=$?
+#Result code 124 means - timed out
+if [ $RES_CODE -ne 0 ]; then
+        ERR_CODE=$RES_CODE
+        if [ $ERR_CODE -eq 124 ]; then
+                echo "Communication with the peer ends with return code: $ERR_CODE (timeout). Warning: this error can mask original error 124 from the transportation process!" >&2
+        else
+                echo "Communication with the peer ends with return code: $ERR_CODE" >&2
+        fi
+        exit $ERR_CODE
 else
-	echo "Communication with the peer ends with return code: $ERR_CODE" >&2
+        echo "Communication with the peer ends with return code: $RES_CODE"
 fi
```

Comment:
Ok, this seems to be a little complicated. What about to change code this way:
```
ERR_CODE=$?
if [ "$ERR_CODE" -eq 0 ]; then
  echo "... file was processed ok ..."
elif [ "$ERR_CODE" -eq 124 ]; then
  echo "...problem with timeout ..."
 exit $ERR_CODE
else
  echo "...any other problem ..."
  exit $ERR_CODE
fi
```
The same for both parts and at the last row "exit $ERR_CODE"
I don't see any reason to use two variables, cause you have only three possible options:

1. users command ended with an error  (exit)
2. groups command ended with an error (users are ok, groups are exiting)
3. both commands are ok (exit 0 at the end of the script)

What do you think? Do I miss something?